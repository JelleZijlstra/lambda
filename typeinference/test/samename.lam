let f = \x.
	data option a = None | Some a in
	let f' = \x. match x with {
		None -> 3
		| Some v -> v + 1
	} in
	(f' None, f' (Some x))
in
let g = \x.
	data option a b = None a | Some b in
	let f' = \x. match x with {
		Some _ -> 4
		| None v -> v - 1
	} in
	(f' (Some true), f' (None x))
in
((f, g), (f 3, g 4))
