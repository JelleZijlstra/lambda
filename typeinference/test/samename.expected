%%stdout%%
Result: module
	let __result__ = ((3, 4), (4, 3))
end
%%stderr%%
